claim/claim.go

```diff
@@ -35,21 +35,22 @@ const (
 // provide the necessary data to upgrade, uninstall, and downgrade
 // a CNAB package.
 type Claim struct {
-	Name       string                 `json:"name"`
-	Revision   string                 `json:"revision"`
-	Created    time.Time              `json:"created"`
-	Modified   time.Time              `json:"modified"`
-	Bundle     *bundle.Bundle         `json:"bundle"`
-	Result     Result                 `json:"result"`
-	Parameters map[string]interface{} `json:"parameters"`
-	Files      map[string]string      `json:"files"`
+	Name          string                    `json:"name"`
+	Revision      string                    `json:"revision"`
+	Created       time.Time                 `json:"created"`
+	Modified      time.Time                 `json:"modified"`
+	Bundle        *bundle.Bundle            `json:"bundle"`
+	Result        Result                    `json:"result"`
+	Parameters    map[string]interface{}    `json:"parameters"`
+	Files         map[string]string         `json:"files"`
+	RelocationMap bundle.ImageRelocationMap `json:"relocationMap"`
 }
 
 // ValidName is a regular expression that indicates whether a name is a valid claim name.
 var ValidName = regexp.MustCompile("^[a-zA-Z0-9_-]+$")
 
 // New creates a new Claim initialized for an installation operation.
-func New(name string) (*Claim, error) {
+func New(name string, relocationMap bundle.ImageRelocationMap) (*Claim, error) {
```

Comment:
I'm a bit on the fence for this, as adding the relocationMap here it is not homogeneous with other fields (like `parameters` or even `bundle`) which are not initialized in this `New` function.
What about removing it from here and let the user set the relocation map herself afterwards?

Reply:
I'm with @silvin-lubecki on this. Unless having a non-empty relocation map is _mandatory_, I would prefer to not require it during construction.

Reply:
I was unsure as well when I added it. From the side we only call `New` on install so I assume that we could piggy back on it. Nevertheless it is a better patter to do not have it there.

Just updated the PR and remove the map from the `New` method